networkit/cpp/algebraic/Vector.cpp

```diff
@@ -37,36 +37,29 @@ double Vector::length() const {
 }
 
 double Vector::mean() const {
-    double sum = 0.0;
-    this->forElements([&](double value){
-        sum += value;
-    });
+    double sum = 0.;
+    const auto n = getDimension();
+#ifndef NETWORKIT_OMP2
+#pragma omp parallel for reduction(+ : sum)
+    for (omp_index i = 0; i < static_cast<omp_index>(n); ++i)
+        sum += values[i];
+#else
+    sum = std::accumulate(values.begin(), values.end(), 0.);
+#endif
 
-    return sum / (double) this->getDimension();
+    return sum / (double)n;
 }
 
 bool Vector::operator==(const Vector &other) const {
-    if (getDimension() != other.getDimension() || isTransposed() != other.isTransposed()) return false;
-
-    for (index i = 0; i < getDimension(); i++) {
-        if (values[i] != other[i]) return false;
-    }
-
-    return true;
+    return values == other.values;
```

Comment:
Hm, it is not 100% the same - but in principle we have here the same problem (also before the proposed change) with comparison of doubles (see #104). 

This would be more expensive, but we could add an epsilon-comparison: https://stackoverflow.com/questions/17333/what-is-the-most-effective-way-for-float-and-double-comparison